rust/noosphere-gateway/src/nns.rs

```diff
@@ -33,39 +32,36 @@ pub enum NameSystemConfiguration {
 }
 
 #[derive(Display)]
-pub enum NameSystemJob<K, S>
-where
-    K: KeyMaterial + Clone + 'static,
-    S: Storage,
-{
+pub enum NameSystemJob<C> {
     /// Resolve all names in the sphere at the latest version
     ResolveAll {
-        context: Arc<Mutex<SphereContext<K, S>>>,
+        context: C,
     },
     /// Resolve a single name from a given sphere at the latest version
     #[allow(dead_code)]
     ResolveImmediately {
-        context: Arc<Mutex<SphereContext<K, S>>>,
+        context: C,
         name: String,
         tx: Sender<Option<Cid>>,
     },
     /// Resolve all added names of a given sphere since the given sphere
     /// revision
     ResolveSince {
-        context: Arc<Mutex<SphereContext<K, S>>>,
+        context: C,
         since: Option<Cid>,
     },
     Publish {
-        context: Arc<Mutex<SphereContext<K, S>>>,
+        context: C,
         record: Jwt,
     },
 }
 
-pub fn start_name_system<K, S>(
+pub fn start_name_system<H, K, S>(
```

Comment:
nit consider throughout: consistently use the same generic (e.g. `C` or `Ctx` maybe) for bounded `H: HasMutableSphereContext` and unbounded (`enum NameSystemJob<C>`) forms, and consider adding bounds to unbounded (`OnDemandNameResolver<C>`) forms

Reply:
I'll standardize the letter, but adding bounds to those other forms is very verbose (because the extra generics need to be expressed in terms of `PhantomData` and/or unusable enum variants). Also, it's unnecessary because the bounds are enforced implicitly (e.g., but the associated `Sender`).

Reply:
I see!

Reply:
Every `H` is now a `C`.